Documentation/ceph-disaster-recovery.md

```diff
@@ -231,9 +237,97 @@ kubectl -n rook-ceph scale deployment rook-ceph-operator --replicas=1
 
 The operator will automatically add more mons to increase the quorum size again, depending on the `mon.count`.
 
-# Adopt an existing Rook Ceph cluster into a new Kubernetes cluster
+## Restoring CRDs After Deletion
+
+When the Rook CRDs are deleted, the Rook operator will respond to the deletion event to attempt to clean up the cluster resources.
+If any data appears present in the cluster, Rook will refuse to allow the resources to be deleted since the operator will
+refuse to remove the finalizer on the CRs until the underlying data is deleted. For more details, see the
+[dependency design doc](https://github.com/rook/rook/blob/master/design/ceph/resource-dependencies.md).
+
+While it is good that the CRs will not be deleted and the underlying Ceph data and daemons continue to be
+available, the CRs will be stuck indefinitely in a `deleting` state in which the operator will not
+continue to ensure cluster health. Upgrades will be blocked, further updates to the CRs are prevented, and so on.
+Since Kubernetes does not allow undeleting resources, the following procedure will allow you to restore
+the CRs to their prior state without even necessarily suffering cluster downtime.
```

Comment:
I'm not sure this is the best design to stop ensuring cluster health. If Rook is blocking cluster deletion, it still needs to verify health of the cluster and take steps to keep it healthy so that other resources (like a CephFilesystem, for example) can continue to be healthy if users need to back up that data (which could take a few days in the worst case I might imagine) before the resource can be deleted.

We can certainly block upgrades and CR modifications that are not the cleanup policy.

Reply:
Why would we continue to reconcile a deleted CR? The desired state is to delete it. If a CR has been deleted, the current behavior of our Rook controllers is to check for resources, then remove the finalizer when safe and there is no other reconciling of an updated CR. If you actually don't want to delete the cluster, you would then go through this procedure to restore it. 

I would expect the state of the cluster is the same as if the operator is down, where no reconciling can be expected.

Reply:
Hmm... I suppose that is a good point. If the deletion is accidental, it will be protected, but it won't continue reconciling. If the user would need it to keep being healthy (like for user data backup) they can use this procedure. It's more steps, but it shouldn't affect many users I would hope.